src/main/kotlin/no/nb/mlt/wls/product/service/ProductService.kt

```diff
@@ -47,7 +48,13 @@ class ProductService(val db: ProductRepository, val synqProductService: SynqProd
         val product = payload.toProduct().copy(quantity = 0.0, location = null)
 
         // Product service should create the product in the storage system, and return error message if it fails
-        if (synqProductService.createProduct(product.toSynqPayload()).statusCode.isSameCodeAs(HttpStatus.OK)) {
+        val synqResponse = synqProductService.createProduct(product.toSynqPayload())
+        // If SynQ didn't throw an error, but returned 4xx/5xx, then it is likely some error or edge-case we haven't handled
+        if (synqResponse.statusCode.is4xxClientError || synqResponse.statusCode.is5xxServerError) {
+            throw ServerErrorException("Unexpected error with SynQ", null)
+        }
```

Comment:
This code might be redundant, as the WebClient should always map a 4xx or 5xx to an error, which is thrown

Reply:
If that's so, we might use it to handle 3xx codes and/or codes other than 200 and 201?

Reply:
I think codes other than what we expect (200 and 201) should be good